src/connectors/snowflake/trulens/connectors/snowflake/utils/server_side_evaluation_artifacts.py

```diff
@@ -0,0 +1,221 @@
+        if not re.match(r"^[A-Za-z0-9_]+$", name):
+            raise ValueError(
+                f"`{error_message_variable_name}` must contain only alphanumeric and underscore characters!"
+            )
+
+    def set_up_all(self) -> None:
+        self._set_up_stage()
+        self._set_up_stream()
+        self._set_up_stored_procedure()
+        self._set_up_wrapper_stored_procedure()
+        self._set_up_task()
+
+    @staticmethod
+    def _remove_leading_spaces(q: str) -> str:
+        # Get leading whitespace in first non-empty line.
+        length_of_leading_whitespace = 0
+        leading_whitespace_in_first_non_empty_line = ""
+        for line in q.split("\n"):
+            if line.strip():
+                length_of_leading_whitespace = len(line) - len(line.lstrip())
+                leading_whitespace_in_first_non_empty_line = line[
+                    :length_of_leading_whitespace
+                ]
+                break
+        # Remove the leading whitespace in the first non-empty line from all non-empty lines.
+        ret = []
+        for line in q.split("\n"):
+            if line.startswith(leading_whitespace_in_first_non_empty_line):
+                ret.append(line[length_of_leading_whitespace:])
+            elif not line.strip():
+                ret.append("")
+            else:
+                raise ValueError()
+        return "\n".join(ret).strip()
+
+    def _run_query(self, q: str) -> None:
+        # Before running the query we first remove leading spaces for two reasons:
+        # 1. Python code in a stored procedure needs to have correct indenting.
+        # 2. To format the query for easier readability in debugging scenarios.
+        self._session.sql(self._remove_leading_spaces(q)).collect()
+
+    def _set_up_stage(self) -> None:
+        self._run_query(f"CREATE STAGE IF NOT EXISTS {_STAGE_NAME}")
+        data_directory = os.path.join(
+            os.path.dirname(__file__), "../../../data/snowflake_stage_zips"
+        )
+        for zip_to_upload in _ZIPS_TO_UPLOAD:
+            self._session.file.put(
+                os.path.join(data_directory, zip_to_upload),
+                f"@{_STAGE_NAME}",
+            )
+
+    def _set_up_stream(self) -> None:
+        self._run_query(
+            f"""
+            CREATE STREAM IF NOT EXISTS {_STREAM_NAME}
+                ON TABLE {self._database_name}.{self._schema_name}.TRULENS_FEEDBACKS
+                SHOW_INITIAL_ROWS = TRUE
+            """
+        )
+
+    def _set_up_stored_procedure(self) -> None:
+        imports = ",".join([
+            f"'@{_STAGE_NAME}/{curr}'" for curr in _ZIPS_TO_UPLOAD
+        ])
+        self._run_query(
+            f"""
+            CREATE PROCEDURE IF NOT EXISTS {_STORED_PROCEDURE_NAME}()
+                RETURNS STRING
+                LANGUAGE PYTHON
+                RUNTIME_VERSION = '3.11'
+                PACKAGES = (
+                    -- TODO(dkurokawa): get these package versions automatically.
```

Comment:
yeah that's unclear to me so far. It's somewhat messy to get the dependencies of the trulens packages right now. This will come in a separate PR. For the time being, I'm hoping there's not going to be really many changes to the list.